### PR TITLE
Model3 (finetuning)

### PR DESCRIPTION
**Pull request by teresaliau**
Added script that fine-tunes MiniLM embeddings on the news dataset using three contrastive losses (Triplet, SimCSE, Cosine). After each model is trained, it generates embeddings for the test set, and saves them as **npz**. 

to run it requires env to have a GEMINI_API_KEY= 

Will generate local outputs (models + embeddings. These outputs **will be ignored** when pushing (stated in .gitignore)

Outputs:
finetuned_models
- finetuned_models/triplet/
- finetuned_models/simcse/
- finetuned_models/cosine/
embeddings_output
- embeddings_output/triplet_embeddings.npz
- embeddings_output/simcse_embeddings.npz
- embeddings_output/cosine_embeddings.npz

current parameteres used (not yet hyperparamter-tuned)

Training Example Creation:
n_samples: 1000 (about 10% of training daaset)

Training:
batch_size: 16 (62.5 batches per epoch, i chose the standard for finetuning sentence-transformers)
epochs: 3 (i chose the standard as well)
warmup_steps: 100 (might decrease this... maybe during hyperparamter-tuning?)
shuffle: True (required to ensure diverse in-batch negatives)

Inference:
batch_size: 32 (i put a larger batch size (higher than training to make it faster at inference))

Reproducibility:

SEED: 42 (Standard choice for reproducibility across runs)

### CODE REVIEW

**Comment by teresaliau** [comment]
Editing the script again
